### ISSUE TITLE
Single call wrappers for ScoreVisualizers

### ISSUE THREAD

**Issue by rebeccabilbro** [DistrictDataLabs/yellowbrick#78]
Implement single call wrappers for ScoreVisualizers and refine for FeatureVisualizers as necessary.

**Comment by rebeccabilbro** [comment]
The idea is to wrap the __init__/predict-score/fit-transform/poof sequence so that users can accomplish a single visualization in a single call (e.g. for a one off visualization). Undecided if this should return the visualizer object or it should return the matplotlib axes. Some examples below:

```python
class MyFeatureVisualizer(FeatureVisualizer):
    ... 

def myfv(X, y=None, **kwargs):
    visualizer = MyFeatureVisualizer(**kwargs)
    visualizer.fit_transform(X, y=None) 
    visualizer.finalize() 
    return visualizer.ax
```
Here, because of #76 we don't have to worry about poof and can directly call finalize. Alternatively we can keep poof the finalize method and call poof(show=False) on the visualizer.

Another example:
```python
class MyScoreVisualizer(ScoreVisualizer):
    ... 

def mysv(estimator, X, y, **kwargs):
    visualizer = MyScoreVisualizer(estimator, **kwargs)

    # Create the train/test split for evaluation 
    Xr, yr, Xt, yt = train_test_split(X, y) 

    # Fit and score the visualizer 
    visualizer.fit(Xr, yr)
    visualizer.score(Xt, yt) 

    visualizer.finalize() 
    return visualizer.ax 
```
These methods are quick one-off visualizations for users (Scikit-Learn has stuff like this) -- though we'd prefer that they use the class interface directly, particularly with Pipelines. We can do stuff like choose the best visualizer for use in the function (e.g. parallel coordinates with feature order optimization) similar to how NLTK chooses the best tokenizer in nltk.word_tokenize but still exposes every class behind the scenes.

They also pull a secondary duty by documenting how exactly we believe the visualizers should be interacted with. For example some ScoreVisualizer objects require train/test splits and cross validation, while others do not.

**Comment by rebeccabilbro** [comment]
See additional dialogue in closed issue #75

**Comment by rebeccabilbro** [comment]
Implemented for :

- [ ] [ClassificationReport](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier.py#L196)
- [ ] [ROCAUC](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier.py#L356)
- [ ] [ClassBalance](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/classifier.py#L526)
- [ ] [PredictionError](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/regressor.py#L164)
- [ ] [ResidualsPlot](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/regressor.py#L348)
